tests/foreman/api/test_hostgroup.py

```diff
@@ -148,6 +148,64 @@ def test_bz_1107708(self):
         self.assertEqual(host_attrs['all_puppetclasses'][0]['name'], 'ntp')
 
 
+@skip_if_bug_open('bugzilla', 1235377)
+class MissingAttrTestCase(APITestCase):
+    """Tests to see if the server returns the attributes it should."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create a ``HostGroup`` as ``cls.host_group``."""
+        cls.host_group = entities.HostGroup().create()
+
+    def test_get_content_source(self):
+        """@Test: Read a host group. Inspect the server's response.
+
+        @Assert: The response contains either a ``content_source`` or
+        ``content_source_id`` attribute.
+
+        @Feature: HostGroup
+
+        """
+        wanted_attrs = ('content_source', 'content_source_id')
```

Comment:
So the difference (other than calling a private function) here is that `read_json()` would return more data than you actually need for the test you want to perform, whereas `_get_entity_id` would return only the forms that you are interested in checking... correct? If so, even though `_get_entity_id` may speed things up a bit (probably not enough to make it worthwhile), `read_json()` is imho more intuitive to follow.